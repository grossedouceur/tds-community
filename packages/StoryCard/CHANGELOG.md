# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.0.1](https://github.com/telus/tds-community/compare/@tds/community-story-card@1.0.0...@tds/community-story-card@1.0.1) (2021-06-29)


### Bug Fixes

* **community-story-card:** add media queries ([ba19936](https://github.com/telus/tds-community/commit/ba19936b830bc82259ad5792bffbd94733c4f999))





# 1.0.0 (2021-06-22)


### Features

* **community-story-card:** add functionality and design for storycard ([85a2a84](https://github.com/telus/tds-community/commit/85a2a84b363037bc7d1971a64958de1cb5d77db1))
