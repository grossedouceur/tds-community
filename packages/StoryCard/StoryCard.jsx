import React from 'react'
import PropTypes from 'prop-types'
import { safeRest } from '@tds/util-helpers'
import FlexGrid from '@tds/core-flex-grid'
import Heading from '@tds/core-heading'
import Paragraph from '@tds/core-paragraph'

const StoryCard = ({
  storyType,
  date,
  title,
  description,
  categoryName,
  imgUrl,
  slug,
  ...rest
}) => {
  const media = {
    type: 'image',
    content: <img src={imgUrl} alt={description} width="100%" />,
  }
  const isImage = media && media.type === 'image'
  const renderImage = img => {
    const imageContainer = <div>{img}</div>
    return imageContainer
  }
  return (
    <FlexGrid.Col xs={12} md={6} {...safeRest(rest)}>
      <div key={title}>
        <Paragraph>{storyType}</Paragraph>
        <Paragraph>{date}</Paragraph>
        <Heading level="h3">{title}</Heading>
        <Paragraph>{description}</Paragraph>
        <Paragraph>{categoryName}</Paragraph>
        <a href={slug}>{isImage && renderImage(media.content)}</a>
      </div>
    </FlexGrid.Col>
  )
}

StoryCard.propTypes = {
  /**
   * The title of each story
   */
  title: PropTypes.string,
  /**
   * The description of each story
   */
  description: PropTypes.string,
  /**
   * The URL for the image to be displayed
   */
  imgUrl: PropTypes.string,
  /**
   * The slug for the story
   */
  slug: PropTypes.string,
  /**
   * The type of story
   */
  storyType: PropTypes.string,
  /**
   * The publishing date
   */
  date: PropTypes.string,
  /**
   * The category
   */
  categoryName: PropTypes.string,
}

StoryCard.defaultProps = {
  title: undefined,
  description: undefined,
  imgUrl: undefined,
  slug: undefined,
  storyType: undefined,
  date: undefined,
  categoryName: undefined,
}

export default StoryCard
