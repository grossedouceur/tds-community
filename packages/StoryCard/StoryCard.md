(describe what your component does here)

### Usage criteria

- What must your experience achieve in order to warrant the use of this component?
- When using this component, how should you typically use it?

```jsx
<StoryCard
  storyType="Article"
  date="May 11th, 2021"
  title="Crisis Text Line provides mental health support for youth during lockdown"
  description="With the help of a Foundation grant, Kids Help Line was able to expand its mental health resources to include a Crisis Text Line in greater Edmonton."
  categoryName="Giving Back"
  imgUrl="blog-example.jpg"
  slug="kids-help-line"
/>
```
