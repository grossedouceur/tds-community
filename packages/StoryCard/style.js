import styled from 'styled-components'
import { colorGreyGainsboro, colorTelusPurple, colorWhite } from '@tds/core-colours'

export const StyledCard = styled.div`
  max-width: 29.188rem;
  min-height: 38.875rem;
  background: ${colorWhite};
  border-radius: 3px;
  border: 1px solid ${colorGreyGainsboro};
  box-shadow: 0 0 1rem #d8d8d8;
  transition: transform 0.4s;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  &:hover {
    transform: scale(103%);
    border: 0.125rem solid ${colorTelusPurple};
  }
  @media (max-width: 414px) {
    max-width: 20rem;
    min-height: 33rem;
  }
  @media (max-width: 320px) {
    max-width: 18rem;
    min-height: 32rem;
  }
`

export const StyledLink = styled.a`
  text-decoration: none;
  color: inherit;
`

export const StyledTextBox = styled.div`
  padding: 0 1rem;
`

export const StyledInfoBox = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 2rem 0;
  @media (max-width: 414px) {
    padding: 1.5rem 0 2rem 0;
  }
  @media (max-width: 320px) {
    padding: 1.5rem 0 1rem 0;
  }
`

export const StyledType = styled.p`
  color: ${colorTelusPurple};
  font-size: 0.875rem;
`

export const StyledDate = styled.p`
  font-size: 0.875rem;
`

export const StyledHeading = styled.h2`
  font-size: 1.219rem;
  font-weight: 400;
  line-height: 1.4;
`

export const StyledHeadingBox = styled.div`
  padding-bottom: 1.5rem;
  @media (max-width: 320px) {
    padding-bottom: 1rem;
  }
`

export const StyledIconBox = styled.div`
  padding-right: 0.5rem;
`

export const StyledImageContainer = styled.div``

export const StyledDescription = styled.p`
  font-size: 0.875rem;
  padding-bottom: 1.5rem;
  @media (max-width: 414px) {
    padding-bottom: 1rem;
  }
`

export const StyledIconTypeBox = styled.div`
  display: flex;
  border-bottom: 1px solid ${colorTelusPurple};
  padding-bottom: 1rem;
  @media (max-width: 414px) {
    padding-bottom: 0.5rem;
  }
`

export const StyledImage = styled.img`
  display: block;
  border-radius: 0 0 2px 2px;
  width: 100%;
`
