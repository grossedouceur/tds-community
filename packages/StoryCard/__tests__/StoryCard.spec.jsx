import React from 'react'
import { shallow } from 'enzyme'

import StoryCard from '../StoryCard'

describe('StoryCard', () => {
  const doShallow = (props = {}) => shallow(<StoryCard {...props} />)

  it('renders', () => {
    const storyCard = doShallow()

    expect(storyCard).toMatchSnapshot()
  })

  it('passes additional attributes to the element', () => {
    const storyCard = doShallow({ id: 'the-id', 'data-some-attr': 'some value' })

    expect(storyCard).toHaveProp('id', 'the-id')
    expect(storyCard).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const storyCard = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(storyCard).not.toHaveProp('className', 'my-custom-class')
    expect(storyCard).not.toHaveProp('style')
  })
})
