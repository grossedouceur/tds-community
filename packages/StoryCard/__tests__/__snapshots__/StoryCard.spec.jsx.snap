// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryCard renders 1`] = `
<Col
  md={6}
  xs={12}
>
  <div>
    <Paragraph
      align="left"
      bold={false}
      invert={false}
      size="medium"
    />
    <Paragraph
      align="left"
      bold={false}
      invert={false}
      size="medium"
    />
    <Heading
      invert={false}
      level="h3"
    />
    <Paragraph
      align="left"
      bold={false}
      invert={false}
      size="medium"
    />
    <Paragraph
      align="left"
      bold={false}
      invert={false}
      size="medium"
    />
    <a>
      <div>
        <img
          width="100%"
        />
      </div>
    </a>
  </div>
</Col>
`;
